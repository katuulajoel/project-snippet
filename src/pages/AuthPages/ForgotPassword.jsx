import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { useHistory } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";

import { Cta } from "../../components/Form/Form";
import Input from "../../components/Input";
import Button from "../../components/Button";
import Error from "../../components/Error";
import { resetPassword } from "../../actions/AuthActions";
import Progress from "../../components/Progress";
// import Success from "../../components/Success";
import MetaTags from "../../components/MetaTags";
import AuthLayout, {
  AuthStylingLayoutChildren,
} from "../../layouts/AuthLayout";

const ForgotPassword = () => {
  const history = useHistory();
  const dispatch = useDispatch();
  const { user, isMakingRequest, errors } = useSelector(({ Auth }) => Auth);
  const [email, setEmail] = useState("");

  useEffect(() => {
    if (user?.id) {
      history.push("/dashboard");
    }
  }, []);

  const onFormSubmit = (e) => {
    e.preventDefault();
    if (!email) {
      return;
    }

    dispatch(resetPassword({ email }));
  };

  return (
    <div className="auth-page">
      <MetaTags
        title={"Login"}
        description={
          "Login to hire skilled African developers ready work on your software project."
        }
      />

      <AuthLayout>
        <AuthStylingLayoutChildren>
          <div>
            <form onSubmit={onFormSubmit} name="login" role="form">
              {errors && errors.resetPassword && (
                <Error
                  message={
                    errors.resetPassword ||
                    "Sorry, we couldn't reset your password. Please try again."
                  }
                />
              )}
              <h3 className="AuthForm__title">Reset Password</h3>
              {isMakingRequest.resetPassword && <Progress />}
              {/* {auth.isReset && (
                <Success message="Instructions for resetting your password have been sent to your email if we find an account associated with it." />
              )} */}
              <div className="form-group">
                <label className="Auth_label">
                  Email address
                  <span>*</span>
                </label>
                <Input
                  label=""
                  type="email"
                  id="email"
                  className="form-control"
                  name="email"
                  value={email || ""}
                  onChange={(e) => setEmail(e.target.value)}
                  placeholder="Enter email address"
                />
              </div>
              <Cta
                style={{
                  textAlign: "right",
                  display: "block",
                  marginTop: "8px",
                }}
                href="/login"
              >
                Have an account? Login
              </Cta>
              <div className="text-center">
                <Button type="submit" disabled={isMakingRequest.resetPassword}>
                  Reset Password
                </Button>
              </div>
            </form>
          </div>
        </AuthStylingLayoutChildren>
      </AuthLayout>
    </div>
  );
};

ForgotPassword.propTypes = {
  isAuthenticated: PropTypes.func,
};

export default ForgotPassword;
