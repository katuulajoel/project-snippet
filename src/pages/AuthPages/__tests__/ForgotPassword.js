import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import ForgotPassword from "../ForgotPassword";
import { mount } from "enzyme/build";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";

const middlewares = [thunk];

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const mockAppState = {
  Auth: {
    user: { uid: 123, email: "test@gmail.com" },
    isMakingRequest: {},
  },
};

const mockAppStore = (state) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state || mockAppState);
};

describe("Login snapshot test", () => {
  it("Snapshot test for Login component", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Provider store={mockAppStore()}>
            <ForgotPassword />
          </Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call resetPassword function", async () => {
    const store = mockAppStore();
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <ForgotPassword />
        </Provider>
      </BrowserRouter>
    );

    wrapper
      .find("#email")
      .at(1)
      .simulate("change", { target: { value: "test@gmail.com" } });

    wrapper.find("button[type='submit']").simulate("submit");

    const expectedActions = [
      { type: "RESET_PASSWORD_START", email: { email: "test@gmail.com" } },
    ];
    expect(store.getActions()).toEqual(expectedActions);
  });

  it("should not call resetPassword function if no email", async () => {
    const store = mockAppStore();
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <ForgotPassword />
        </Provider>
      </BrowserRouter>
    );

    wrapper.find("button[type='submit']").simulate("submit");
    expect(store.getActions()).toEqual([]);
  });

  it("should redirect if authenticated", async () => {
    render(
      <MemoryRouter>
        <Provider
          store={mockAppStore({
            Auth: { user: { id: 123456 }, isMakingRequest: {} },
          })}
        >
          <ForgotPassword />
        </Provider>
      </MemoryRouter>
    );
    expect(mockHistoryPush).toHaveBeenCalledWith("/dashboard");
  });
});
