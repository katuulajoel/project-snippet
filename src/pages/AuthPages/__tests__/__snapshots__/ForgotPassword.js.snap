// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot test Snapshot test for Login component 1`] = `
<div
  className="auth-page"
>
  <div
    className="sc-hKFxyN cMRkkx"
  >
    <div
      className="grid_layout d-block d-md-grid"
    >
      <div
        className="grid-1 d-flex"
      >
        <div
          className="child__item container"
          data-testid="auth-layout"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            <img
              className="mb-4 logo"
              src={Object {}}
            />
          </a>
          <div
            className="sc-eCApnc kSuWvC"
          >
            <div>
              <form
                name="login"
                onSubmit={[Function]}
                role="form"
              >
                <h3
                  className="AuthForm__title"
                >
                  Reset Password
                </h3>
                <div
                  className="form-group"
                >
                  <label
                    className="Auth_label"
                  >
                    Email address
                    <span>
                      *
                    </span>
                  </label>
                  <div
                    className="sc-gtsrHT ccqUgE"
                    data-testid="input"
                  >
                    <input
                      className="form-control"
                      data-testid=""
                      id="email"
                      label=""
                      name="email"
                      onChange={[Function]}
                      placeholder="Enter email address"
                      type="email"
                      value=""
                    />
                  </div>
                </div>
                <a
                  className="sc-bdnxRM cdRsdd  Form__cta"
                  href="/login"
                  onClick={[Function]}
                  style={
                    Object {
                      "display": "block",
                      "marginTop": "8px",
                      "textAlign": "right",
                    }
                  }
                >
                  Have an account? Login
                </a>
                <div
                  className="text-center"
                >
                  <button
                    className="btn btn-default   "
                    form={null}
                    type="submit"
                  >
                    Reset Password
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        className="grid-2 d-none d-md-block"
        style={
          Object {
            "backgroundImage": "url([object Object])",
            "backgroundSize": "cover",
          }
        }
      />
    </div>
  </div>
</div>
`;
